### PR TITLE
Error on wasm worker creation when SAB is not available

### PR DESCRIPTION
**Pull request by sbc100**
This should have been part of #22710

### CODE REVIEW

File: src/library_wasm_worker.js [hunk 1]
```diff
@@ -156,6 +155,12 @@ if (ENVIRONMENT_IS_WASM_WORKER
   addEventListener("message", _wasmWorkerAppendToQueue);
 }`,
   _emscripten_create_wasm_worker: (stackLowestAddress, stackSize) => {
+    if (typeof SharedArrayBuffer == 'undefined') {
```
[the remark this answers is not in the document]
Comment (sbc100) [reply src/library_wasm_worker.js:158]: We do this with pthreads, but we also include a call to it on pthread create: https://github.com/emscripten-core/emscripten/blob/40e6b046cf7d3d9d0789196a3e26689ccaec0948/src/library_pthread.js#L699-L704

I assume you would like *not* to include that check here?   Or could we include it when ASSERTIONS are enabled perhaps?

[the remark this answers is not in the document]
Comment (sbc100) [reply src/library_wasm_worker.js:158]: Also, since we already have `emscripten_has_threading_support` perhaps we don't need a new function.

**Review by kripken — APPROVED** [review]

File: src/library_wasm_worker.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (sbc100) [reply src/library_wasm_worker.js]: Should we do this same for pthreads perhaps?

**Comment by sbc100** [comment]
Lets land #22723 separately.

File: src/library_wasm_worker.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (kripken) [reply src/library_wasm_worker.js]: Sounds good.

**Pull request merged**